python_ta/checkers/forbidden_import_checker.py

```diff
@@ -77,6 +86,10 @@ def visit_importfrom(self, node: nodes.ImportFrom) -> None:
             node.modname not in self.linter.config.allowed_import_modules
             and node.modname not in self.linter.config.extra_imports
             and node.modname not in self.get_allowed_local_files()
+            and not all(
```

Comment:
The overall logic here is good (though see my above comment about consolidating options), however I would indeed like to see the error message improved and be more specific, and so you can't just use an `all`.

First, the original message was misleading (`"You may not import any modules - you imported %s on line %s."`) because it's not that students aren't allowed to import any modules, just that they are restricted in what they can import.

Instead, please modify the message to just `"You may not import %s"` (removing the line number---since that shows in the context anyway), where `%s` can be replaced by either `"module <x>"` for a forbidden module, or `"<y> from module <x>"` if it's a specific name. Ideally `<y>` would be a comma-separated list of names, in the case that there are multiple imports on the same line that aren't allowed.